src/Codeception/Module/WebDriver.php

```diff
@@ -2540,22 +2540,62 @@ public function switchToWindow($name = null)
      */
     public function switchToIFrame($locator = null)
     {
-        if (is_null($locator)) {
+        $this->findAndSwitchToFrame($locator, 'iframe');
+    }
+
+    /**
+     * Switch to another frame on the page.
+     *
+     * Example:
+     * ``` html
+     * <frame name="another_frame" id="fr1" src="http://example.com">
+     *
+     * ```
+     *
+     * ``` php
+     * <?php
+     * # switch to frame by name
+     * $I->switchToFrame("another_frame");
+     * # switch to frame by CSS or XPath
+     * $I->switchToFrame("#fr1");
+     * # switch to parent page
+     * $I->switchToFrame();
+     *
+     * ```
+     *
+     * @param string|null $locator (name, CSS or XPath)
+     */
+    public function switchToFrame($locator = null)
+    {
+        $this->findAndSwitchToFrame($locator, 'frame');
+    }
+
+    /**
+     * @param string|null $locator
+     * @param string $tag
+     */
+    private function findAndSwitchToFrame($locator = null, $tag = 'frame')
+    {
+        if ($locator === null) {
             $this->webDriver->switchTo()->defaultContent();
             return;
         }
+
         try {
-            $els = $this->_findElements("iframe[name='$locator']");
```

Comment:
I think that it would be better to make `switchToIFrame` behave as before by changing locator to match frames and iframes:
```php
$els = $this->_findElements("iframe[name='$locator']  | frame[name='$locator']");
```

Reply:
I would not say so, as the name of method is missleading, it have IFrame in it, so it should work with iframes, even if it was possible to be used with frames before. Or, the name should be changed to switchToFrame then, and used for iframes as well. But thats just my opinion, would be better to vote for it probably :)

Reply:
Changing the name is backwards compatibility break, 
making switchToIframe work with frames as before is backwards compatibility fix.

Reply:
Yes, but:
* Making it working as before will make the method misused again. The fix should be introduced, but with the deprecation of this functionality maybe?
* Renaming could be done with the deprecation of the old method as well, it's not critical to just rename it.

Sure, it is your decision how to handle this question, but me personally would prefer the chaos to be more organized with the time, but not just fixed to be working somehow :)

Reply:
@DavertMik thinks that switchToIframe working with frames is a feature, not a bug.

Reply:
@Naktibalda But it leads to misunderstandings while it does more than its described, and then people can change the code without even thinking about those "features" and one day those are stopping to work. So either you need two methods or better documentation/implementation on the old one. Feel free to do as you think is better.